omniscidb/QueryBuilder/QueryBuilder.h

```diff
@@ -0,0 +1,656 @@
+
+  BuilderExpr sub(const BuilderExpr& rhs) const;
+  BuilderExpr sub(int val) const;
+  BuilderExpr sub(int64_t val) const;
+  BuilderExpr sub(float val) const;
+  BuilderExpr sub(double val) const;
+
+  BuilderExpr sub(const BuilderExpr& rhs, DateAddField field) const;
+  BuilderExpr sub(int val, DateAddField field) const;
+  BuilderExpr sub(int64_t val, DateAddField field) const;
+  BuilderExpr sub(const BuilderExpr& rhs, const std::string& field) const;
+  BuilderExpr sub(int val, const std::string& field) const;
+  BuilderExpr sub(int64_t val, const std::string& field) const;
+
+  BuilderExpr mul(const BuilderExpr& rhs) const;
+  BuilderExpr mul(int val) const;
+  BuilderExpr mul(int64_t val) const;
+  BuilderExpr mul(float val) const;
+  BuilderExpr mul(double val) const;
+
+  BuilderExpr div(const BuilderExpr& rhs) const;
+  BuilderExpr div(int val) const;
+  BuilderExpr div(int64_t val) const;
+  BuilderExpr div(float val) const;
+  BuilderExpr div(double val) const;
+
+  BuilderExpr mod(const BuilderExpr& rhs) const;
+  BuilderExpr mod(int val) const;
+  BuilderExpr mod(int64_t val) const;
+
+  BuilderExpr logicalAnd(const BuilderExpr& rhs) const;
+  BuilderExpr logicalOr(const BuilderExpr& rhs) const;
+
+  BuilderExpr eq(const BuilderExpr& rhs) const;
+  BuilderExpr eq(int val) const;
+  BuilderExpr eq(int64_t val) const;
+  BuilderExpr eq(float val) const;
+  BuilderExpr eq(double val) const;
+  BuilderExpr eq(const std::string& val) const;
+
+  BuilderExpr ne(const BuilderExpr& rhs) const;
+  BuilderExpr ne(int val) const;
+  BuilderExpr ne(int64_t val) const;
+  BuilderExpr ne(float val) const;
+  BuilderExpr ne(double val) const;
+  BuilderExpr ne(const std::string& val) const;
+
+  BuilderExpr lt(const BuilderExpr& rhs) const;
+  BuilderExpr lt(int val) const;
+  BuilderExpr lt(int64_t val) const;
+  BuilderExpr lt(float val) const;
+  BuilderExpr lt(double val) const;
+  BuilderExpr lt(const std::string& val) const;
+
+  BuilderExpr le(const BuilderExpr& rhs) const;
+  BuilderExpr le(int val) const;
+  BuilderExpr le(int64_t val) const;
+  BuilderExpr le(float val) const;
+  BuilderExpr le(double val) const;
+  BuilderExpr le(const std::string& val) const;
+
+  BuilderExpr gt(const BuilderExpr& rhs) const;
+  BuilderExpr gt(int val) const;
+  BuilderExpr gt(int64_t val) const;
+  BuilderExpr gt(float val) const;
+  BuilderExpr gt(double val) const;
+  BuilderExpr gt(const std::string& val) const;
+
+  BuilderExpr ge(const BuilderExpr& rhs) const;
+  BuilderExpr ge(int val) const;
+  BuilderExpr ge(int64_t val) const;
+  BuilderExpr ge(float val) const;
+  BuilderExpr ge(double val) const;
+  BuilderExpr ge(const std::string& val) const;
+
+  BuilderExpr at(const BuilderExpr& idx) const;
+  BuilderExpr at(int idx) const;
+  BuilderExpr at(int64_t idx) const;
+
+  BuilderExpr rewrite(ExprRewriter& rewriter) const;
+
+  BuilderExpr operator!() const;
+  BuilderExpr operator-() const;
```

Comment:
What does it semantically mean "to subtract two builder expressions"?

Reply:
This is a unary minus. It allows to use it as `neg_val = -node.ref("value")`. There is also a binary operator which allows you to build subtractions like `diff = node["x"] - node["y"]`.